### PR TITLE
SQL: [Docs] Add an ES-SQL column for data types

### PR DESCRIPTION
**Pull request by matriv**
In order to distinguish the ES-SQL type from the standard SQL type
add a new ES-SQL column that will make clear this distingstion,
e.g.: datetime vs TIMSTAMP

Fixes: #37519

### CODE REVIEW

**Comment by matriv** [comment]
Should we backport to earlier versions?

**Review by astefan — APPROVED** [review]
LGTM. Left one minor comment and one open suggestion.

File: docs/reference/sql/language/data-types.asciidoc [hunk 1]
```diff
@@ -7,39 +7,40 @@ beta[]
...
+| <<number, `integer`>>       | integer       | INTEGER     | 10
+| <<number, `long`>>          | long          | BIGINT      | 19
+| <<number, `double`>>        | double        | DOUBLE      | 15
+| <<number, `float`>>         | float         | REAL        | 7
+| <<number, `half_float`>>    | half_float    | FLOAT       | 16
+| <<number, `scaled_float`>>  | scaled_float  | FLOAT       | 19
+| <<keyword, `keyword`>>      | keyword       | VARCHAR     | based on <<ignore-above>>
+| <<text, `text`>>            | text          | VARCHAR     | 2,147,483,647
+| <<binary, `binary`>>        | binary        | VARBINARY   | 2,147,483,647
+| <<date, `date`>>            | datetime      | TIMESTAMP   | 24
```
[the remark this answers is not in the document]
Comment (matriv) [reply docs/reference/sql/language/data-types.asciidoc]: Added the clarification, as I think it's better to have it standing out there apart from the table where the eye can skip it as everything else is the same.

[the remark this answers is not in the document]
Comment (costin) [reply docs/reference/sql/language/data-types.asciidoc]: Potentially add an explanation as why there's datetime : `in order to not confuse it with the ANSI SQL `date` (date only) type`.

**Review by costin — APPROVED** [review]
LGTM

File: docs/reference/sql/language/data-types.asciidoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (matriv) [reply docs/reference/sql/language/data-types.asciidoc]: Yep, planning to do it with the introduction of date.

**Comment by matriv** [comment]
Backported to `6.x` with 40af8a336c83d5c4ae3b806679a20771f6db4a7f

**Pull request merged**